File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        wordGroups = defaultdict(list)
        for word in strs:
            key = [0] * 26

            for c in word:
                key[ord(c) - ord('a')] += 1
            wordGroups[tuple(key)].append(word)

        res = []
        for key in wordGroups:
            res.append(wordGroups[key])
        return res

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {

        std::unordered_map<std::string, vector<std::string>> groups;
        
        for (std::string current_string : strs ){
            std::string key = current_string;
            std::sort(key.begin(), key.end());
            groups[key].push_back(current_string);
        }

        std::vector<std::vector<std::string>> result; 
        
        for (auto key_value : groups){
            result.push_back(key_value.second);
        }
        return result;
    }
};